ui/pages/notifications-settings/notifications-settings.tsx

```diff
@@ -57,7 +57,7 @@ export default function NotificationsSettings() {
   const isUpdatingMetamaskNotifications = useSelector(
     getIsUpdatingMetamaskNotifications,
   );
-  const accounts: AccountType[] = useSelector(getInternalAccounts);
+  const accounts = useSelector(getInternalAccounts) as AccountType[];
```

Comment:
Was wondering why we are moving to a type assertion. Tested locally. This is due to our teams poor typing [here above in this file](https://github.com/MetaMask/metamask-extension/blob/c983298d29f5945592f40ff4d981ea50b7ad2ebc/ui/pages/notifications-settings/notifications-settings.tsx#L42-L46).

I'm tempted to either:
1. Rely on inferred types
2. Narrow the above type - we literally only care if there is an address field!!
```ts
type AccountType = { address: string }
```

Reply:
To keep this PR unblocked, our team can take this as a separate PR.

Reply:
Fixed here: https://github.com/MetaMask/metamask-extension/pull/28271